e2e-tests/src/tests/external-datasets.test.ts

```diff
@@ -341,3 +344,231 @@ test.describe.serial('External Datasets', () => {
+    expect(mission_model_id).toBeDefined();
+    expect(typeof mission_model_id).toEqual("number");
+
+    const plan_input : CreatePlanInput = {
+      model_id : mission_model_id,
+      name : 'test_plan' + rd,
+      start_time : plan_start_timestamp,
+      duration : time.getIntervalFromDoyRange(plan_start_timestamp, plan_end_timestamp)
+    };
+    plan_id = await req.createPlan(request, plan_input);
+    expect(plan_id).not.toBeNull();
+    expect(plan_id).toBeDefined();
+    expect(typeof plan_id).toEqual("number");
+  });
+
+  test("Add Constraint to Plan", async ({ request }) => {
+    const constraint: ConstraintInsertInput = {
+      name: constraint_name,
+      definition: "export default (): Constraint => Discrete.Resource(\"/my_boolean\").equal(true)",
+      description: "",
+      model_id: null,
+      plan_id,
+    };
+    constraint_id = await req.insertConstraint(request, constraint);
+
+    expect(constraint_id).not.toBeNull();
+    expect(constraint_id).toBeDefined();
+  });
+
+  test("Run simulation", async ({ request }) => {
+    const resp: SimulationResponse = await awaitSimulation(request, plan_id);
+    first_simulation_dataset_id = resp.simulationDatasetId;
+
+    expect(resp.status).toEqual("complete");
+  });
+
+  test('Upload Simulation Associated External Dataset', async ({ request }) => {
+    const externalDatasetInput: ExternalDatasetInsertInput = {
+      plan_id,
+      associated_simulation_dataset_id: first_simulation_dataset_id,
+      dataset_start: dataset_start_timestamp,
+      profile_set
+    };
+
+    dataset_id = await req.insertExternalDataset(request, externalDatasetInput);
+    expect(dataset_id).not.toBeNull();
+    expect(dataset_id).toBeDefined();
+    expect(dataset_id).not.toEqual(first_simulation_dataset_id);
+    expect(typeof dataset_id).toEqual("number");
+  });
+
+  test('Query External Dataset', async ({ request }) => {
+    profile_set_result.plan_dataset_by_pk.associated_simulation_dataset_id = first_simulation_dataset_id;
+    const getExternalDatasetInput: ExternalDatasetQueryInput = { plan_id, dataset_id };
+
+    const result = await req.getExternalDataset(request, getExternalDatasetInput);
+
+    expect(result).toEqual(profile_set_result);
+  });
+
+  test("Check there is one violation when simulationDatasetId isn't provided", async ({ request }) => {
+    const violations: ConstraintViolation[] = await req.checkConstraints(request, plan_id);
+
+    expect(violations).not.toBeNull();
+    expect(violations).toBeDefined();
+    expect(violations).toHaveLength(1);
+  });
+
+  test("Check there is one violation when simulationDatasetId is provided", async ({ request }) => {
+    const violations: ConstraintViolation[] = await req.checkConstraints(request, plan_id, first_simulation_dataset_id);
+    violation = violations[0];
+
+    expect(violations).not.toBeNull();
+    expect(violations).toBeDefined();
+    expect(violations).toHaveLength(1);
+  });
+
```

Comment:
It looks like the violation gotten in both of these tests is expected to be the same one. Can that assertion please be added?